src/rules/markUsed.js

```diff
@@ -0,0 +1,59 @@
+import iterateJsdoc from '../iterateJsdoc';
+
+/**
+ * Extracts the type names from a type declaration string.
+ *
+ * @see https://jsdoc.app/tags-type.html
+ */
+const extractTypeNames = (type) => {
+  if (type.startsWith('(') && type.endsWith(')')) {
+    // Type union
+    return type.slice(1, type.length - 1).split('|').flatMap(extractTypeNames);
+  } else if (type.endsWith('[]')) {
+    // Arrays
```

Comment:
Rather than parsing the types manually, I would instead ask you to use `jsdoc-type-pratt-parser` to get more refined parsing of types. Any issues with its parsing can be filed with that project.

Reply:
I've now pushed a fix for this.